tests/src/application/middlewares/auth.spec.ts

```diff
@@ -35,111 +35,106 @@ const makeSut = (): SutTypes => {
+  describe('Token Decoder', () => {
+    it('Should call TokenDecoder with correct authorization', async () => {
       const { sut, tokenDecoderStub } = makeSut()
-      const spyDecoder = jest.spyOn(tokenDecoderStub, 'decode')
-
-      const httpRequest = makeFakeAuthorization({ data: 'any_token' })
+      const tokenDecoderSpy = jest.spyOn(tokenDecoderStub, 'decode')
       await sut.handle(httpRequest)
-
-      expect(spyDecoder).toHaveBeenCalledWith(httpRequest.authorization)
+      expect(tokenDecoderSpy).toHaveBeenCalledWith(httpRequest.authorization)
     })
 
-    it('Should return 500 if tokenDecoder throws', async () => {
+    it('Should return 500 (ServerError) if TokenDecoder throws', async () => {
       const { sut, tokenDecoderStub } = makeSut()
-      jest.spyOn(tokenDecoderStub, 'decode').mockRejectedValueOnce(new Error())
-
-      const httpRequest = makeFakeAuthorization({ data: 'any_token' })
+      jest.spyOn(tokenDecoderStub, 'decode').mockRejectedValue(new Error())
       const httpResponse = await sut.handle(httpRequest)
-
       expect(httpResponse).toEqual(makeFakeServerError())
     })
-  })
-
-  describe('test the loadGuardianById service', () => {
-    it('Should return 401 if invalid payload is provided', async () => {
-      const { sut, tokenDecoderStub, loadGuardianByIdStub } = makeSut()
-      jest.spyOn(tokenDecoderStub, 'decode').mockResolvedValueOnce({})
-      jest.spyOn(loadGuardianByIdStub, 'loadById').mockResolvedValueOnce(undefined)
-
-      const httpRequest = makeFakeAuthorization({ data: 'valid_token' })
-      const httpResponse = await sut.handle(httpRequest)
-
-      expect(httpResponse).toEqual(unauthorized())
-    })
 
-    it('Should return 401 if valid payload is provided with invalid userId', async () => {
-      const { sut, tokenDecoderStub, loadGuardianByIdStub } = makeSut()
-      jest.spyOn(tokenDecoderStub, 'decode').mockResolvedValueOnce({ userId: 'invalid_id' })
-      jest.spyOn(loadGuardianByIdStub, 'loadById').mockResolvedValueOnce(undefined)
-
-      const httpRequest = makeFakeAuthorization({ data: 'valid_token' })
+    it('Should return 401 (Unauthorized) if authorization is invalid (expired or malformed)', async () => {
+      const { sut, tokenDecoderStub } = makeSut()
+      jest.spyOn(tokenDecoderStub, 'decode').mockResolvedValue(null)
       const httpResponse = await sut.handle(httpRequest)
-
-      expect(httpResponse).toEqual(unauthorized())
+      expect(httpResponse).toEqual(unauthorized(new InvalidTokenError('Invalid or expired token')))
     })
+  })
 
-    it('Should call loadGuardianById with correct value', async () => {
-      const { sut, loadGuardianByIdStub } = makeSut()
-      const spyLoadGuardianById = jest.spyOn(loadGuardianByIdStub, 'loadById')
-
-      const httpRequest = makeFakeAuthorization({ data: 'valid_token' })
+  describe('LoadByGuardianById', () => {
+    it('Should call LoadByGuardianById with correct userId', async () => {
```

Comment:
In my opinion, the correct case may be the last one. Because when we are developing with TDD usually we start with the errors cases. What do you think about it?

Reply:
I believe this is a more technical choice, but I enjoyed doing it and will implement it.